ibm/resource_ibm_private_dns_glb_pool.go

```diff
@@ -0,0 +1,340 @@
+							Required: true,
+						},
+						pdnsGlbPoolOriginsEnabled: {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						pdnsGlbPoolOriginsDescription: {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			pdnsGlbPoolMonitor: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The ID of the load balancer monitor to be associated to this pool",
+			},
+			pdnsGlbPoolChannel: {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"webhook"}),
+				Description:  "The notification channel",
+			},
+			pdnsGlbPoolRegion: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"us-south", "us-east", "eu-gb", "eu-du", "au-syd", "jp-tok"}),
+				Description:  "Health check region of VSIs",
+			},
+			pdnsGlbPoolSubnet: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Health check subnet IDs of VSIs",
+			},
+			pdnsGlbPoolCreatedOn: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			pdnsGlbPoolModifiedOn: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceIBMPrivateDNSGLBPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	instanceID := d.Get(pdnsInstanceID).(string)
+	CreatePoolOptions := sess.NewCreatePoolOptions(instanceID)
+
+	poolname := d.Get(pdnsGlbPoolName).(string)
+	CreatePoolOptions.SetName(poolname)
+
+	if description, ok := d.GetOk(pdnsGlbPoolDescription); ok {
+		CreatePoolOptions.SetDescription(description.(string))
+	}
+	if enable, ok := d.GetOk(pdnsGlbPoolEnabled); ok {
+		CreatePoolOptions.SetEnabled(enable.(bool))
+	}
+	if threshold, ok := d.GetOk(pdnsGlbPoolHealthyOriginsThreshold); ok {
+		CreatePoolOptions.SetHealthyOriginsThreshold(int64(threshold.(int)))
+	}
+	if monitor, ok := d.GetOk(pdnsGlbPoolMonitor); ok {
+		CreatePoolOptions.SetMonitor(monitor.(string))
+	}
+	if chanel, ok := d.GetOk(pdnsGlbPoolChannel); ok {
+		CreatePoolOptions.SetNotificationChannel(chanel.(string))
+	}
+	if region, ok := d.GetOk(pdnsGlbPoolRegion); ok {
+		CreatePoolOptions.SetHealthcheckRegion(region.(string))
+	}
+	if subnet, ok := d.GetOk(pdnsGlbPoolSubnet); ok {
+		CreatePoolOptions.SetHealthcheckSubnets(subnet.([]string))
+	}
+
+	poolorigins := d.Get(pdnsGlbPoolOrigins).(*schema.Set)
+	CreatePoolOptions.SetOrigins(expandPoolOrigins(poolorigins))
+
+	result, resp, err := sess.CreatePool(CreatePoolOptions)
+	if err != nil {
+		log.Printf("create global load balancer pool failed %s", resp)
+		return err
+	}
+	log.Println(resp)
+	log.Printf("global load balancer pool created successfully : %s", *result.ID)
+	d.SetId(fmt.Sprintf("%s/%s", instanceID, *result.ID))
+	d.Set(pdnsGlbPoolID, *result.ID)
```

Comment:
move this line to `resourceIBMPrivateDNSGLBPoolRead()`

Reply:
Handled

Reply:
@nimisha0092 I don't think it is handled.. I still see this line `d.Set(pdnsGlbPoolID, *result.ID)`